pygmt/datatypes/dataset.py

```diff
@@ -226,11 +228,16 @@ def to_dataframe(
                 pd.Series(data=np.char.decode(textvector), dtype=pd.StringDtype())
             )
 
+        # Return an empty DataFrame if no columns are found.
+        if len(vectors) == 0:
+            return pd.DataFrame()
+
+        # Create a DataFrame object by concatenating multiple columns
         df = pd.concat(objs=vectors, axis="columns")
```

Comment:
Let's set the column_names, and also the dtypes below later, instead of returning an empty `pd.DataFame` directly:

```suggestion
        if len(vectors) == 0:
            df = pd.DataFrame(columns=column_names)
        else:
            # Create a DataFrame object by concatenating multiple columns
            df = pd.concat(objs=vectors, axis="columns")
```